pkg/shares/split_compact_shares.go

```diff
@@ -183,6 +167,40 @@ func (css *CompactShareSplitter) writeDataLengthVarintToFirstShare(dataLengthVar
 	css.shares[0] = newFirstShare
 }
 
+// maybeWriteReservedByteToPendingShare will be a no-op if the reserved byte has
+// already been populated. If the reserved byte is empty, it will write the
+// location of the next unit of data to the reserved byte.
+func (css *CompactShareSplitter) maybeWriteReservedByteToPendingShare() {
```

Comment:
admittedly the naming for this maybe weird, but I also don't have any better options. hopefully whenever we have more time to refactor this we can and just have a method to write the reserved byte